platform/src/main/java/org/stellar/anchor/platform/paymentobserver/StellarPaymentObserver.java

```diff
@@ -0,0 +1,123 @@
+package org.stellar.anchor.platform.paymentobserver;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import org.stellar.anchor.util.Log;
+import org.stellar.sdk.KeyPair;
+import org.stellar.sdk.Server;
+import org.stellar.sdk.requests.EventListener;
+import org.stellar.sdk.requests.PaymentsRequestBuilder;
+import org.stellar.sdk.requests.SSEStream;
+import org.stellar.sdk.responses.operations.OperationResponse;
+import org.stellar.sdk.responses.operations.PaymentOperationResponse;
+import shadow.com.google.common.base.Optional;
+import shadow.com.google.gson.Gson;
+
+public class StellarPaymentObserver {
+  final Server server;
+  final List<PaymentListener> observers;
+  final List<String> accounts;
+  final List<SSEStream<OperationResponse>> streams;
+  final PageTokenStore pageTokenStore;
+
+  StellarPaymentObserver(
+      String horizonServer,
+      List<String> accounts,
+      List<PaymentListener> observers,
+      PageTokenStore pageTokenStore) {
+    this.server = new Server(horizonServer);
+    this.observers = observers;
+    this.accounts = accounts;
+    this.streams = new ArrayList<>(accounts.size());
+    this.pageTokenStore = pageTokenStore;
+  }
+
+  /** Start watching the accounts. */
+  public void start() {
+    for (String account : accounts) {
+      SSEStream<OperationResponse> stream = watch(account);
+      this.streams.add(stream);
+    }
+  }
+
```

Comment:
❗   One streamer per account would create many streamers in parallel and that could reach the rate limit quicker.

The approach we did for Vibrant was to stream the `{horizon}/payment` endpoint and filter the results based on the accounts we were looking for. Then we made the DB changes based on those filtered results.

Also, in Vibrant we managed to reach a rate limit with a single streamer when we used an old cursor with a limit smaller than `50` (not so sure about the exact number, TBH). That means an anchor with multiple assets  (and distribution accounts) could meet the rate limit soon.

💡  We can create an issue just for this change and handle it in separate as a tech debt.

Reply:
I created an issue for this in https://github.com/stellar/java-stellar-anchor-sdk/issues/182.